Design_Doc_Examples/RAG_Q&A_for collaborative_work_platform.md

```diff
@@ -136,16 +136,110 @@ No ideas
     3. Updating validation splits in response to new data or changing distributions is essential for maintaining the relevance and accuracy of performance estimates.
     4. Detailed planning and documentation of the chosen validation schemas within the design document are vital for ensuring the evaluation process is aligned with the project's goals and constraints.
 
-### **V. Baseline Solution**
 
-Regexp/full text search (Elasticsearch) to get Documents and Sentences
-Show to user highlighted parts of document
+### V. Baseline Solution
 
-- **Key Takeaways:**
-    1. Establishing a baseline is a critical first step in machine learning system design, serving as a simple, operational starting point for iterative improvement.
-    2. The choice of baseline should be guided by a trade-off between desired accuracy and the effort required for development, with simplicity often providing significant advantages in terms of robustness, scalability, and interpretability.
-    3. In deep learning applications, leveraging pretrained models or training simple models from scratch can provide effective baselines, with the choice influenced by the specific requirements and constraints of the project.
-    4. Continuous evaluation and comparison against the baseline are essential for guiding the development process, ensuring that complexity is added only when it yields proportional benefits in performance.
+#### Document Extraction Process
+
+Considering the minimal variability among the documents and the sufficient coverage of most cases within the current dataset, it is recommended to implement an in-house document extraction pipeline. This pipeline should consist of:
+
+1. File Type Handler: Differentiate and handle file types accordingly, as PDFs and images may require additional processing steps.
+2. Text Extraction: Deploy a customized OCR solution designed to handle non-text elements.
+3. Text Preprocessing: Remove unwanted characters, whitespace, or any artifacts.
+4. Markdown Formatting: Ensure that the extracted content is formatted correctly according to markdown standards.
+5. Error Management & Spell Checking: Integrate an error handler and a spell checker to maintain data quality.
+
+#### Retrieval-Augmented Generation Framework
+
+The Retrieval-Augmented Generation (RAG) framework can be broken down into two main components:
+
+- Retrieval
```

Comment:
@tyomj do we want to detail here that Baseline won't support Chat history, requesting for more details from clients and other couple points from the problem definition?

Reply:
Good catch. I'll add it, since it hasn't been described anywhere else.